pymc3/step_methods/mlda.py

```diff
@@ -0,0 +1,384 @@
+        # Check if the tuning flag has been changed and if yes,
+        # change the proposal's tuning flag and reset self.accepted
+        # This is triggered by _iter_sample while the highest-level MLDA step
+        # method is running. It then propagates to all levels.
+        if self.proposal_dist.tune != self.tune:
+            self.proposal_dist.tune = self.tune
+            # set tune in sub-methods of compound stepper explicitly because
+            # it is not set within sample.py (only the CompoundStep's tune flag is)
+            if isinstance(self.next_step_method, CompoundStep):
+                for method in self.next_step_method.methods:
+                    method.tune = self.tune
+            self.accepted = 0
+
+        # Convert current sample from numpy array ->
+        # dict before feeding to proposal
+        q0_dict = self.bij.rmap(q0)
+
+        # Call the recursive DA proposal to get proposed sample
+        # and convert dict -> numpy array
+        q = self.bij.map(self.proposal_dist(q0_dict))
+
+        # Evaluate MLDA acceptance log-ratio
+        # If proposed sample from lower levels is the same as current one,
+        # do not calculate likelihood, just set accept to 0.0
+        if (q == q0).all():
+            accept = np.float(0.0)
+            skipped_logp = True
+        else:
+            accept = self.delta_logp(q, q0) + self.delta_logp_next(q0, q)
+            skipped_logp = False
+
+        # Accept/reject sample - next sample is stored in q_new
+        q_new, accepted = metrop_select(accept, q, q0)
+        if skipped_logp:
+            accepted = False
+
+        # Update acceptance counter
+        self.accepted += accepted
+
+        stats = {
+            'tune': self.tune,
+            'accept': np.exp(accept),
+            'accepted': accepted
+        }
+
+        # Capture latest base chain scaling stats from next step method
+        self.base_scaling_stats = {}
+        if isinstance(self.next_step_method, CompoundStep):
+            scaling_list = []
+            for method in self.next_step_method.methods:
+                scaling_list.append(method.scaling)
+            self.base_scaling_stats = {"base_scaling": np.array(scaling_list)}
+        elif not isinstance(self.next_step_method, MLDA):
+            # next method is any block sampler
+            self.base_scaling_stats = {"base_scaling": np.array(self.next_step_method.scaling)}
+        else:
+            # next method is MLDA - propagate dict from lower levels
+            self.base_scaling_stats = self.next_step_method.base_scaling_stats
+        stats = {**stats, **self.base_scaling_stats}
+
+        return q_new, [stats]
+
+    @staticmethod
+    def competence(var, has_grad):
+        """Return MLDA competence for given var/has_grad. MLDA currently works
+        only with continuous variables."""
+        if var.dtype in pm.discrete_types:
+            return Competence.INCOMPATIBLE
+        return Competence.COMPATIBLE
```

Comment:
can you apply [`black`](https://github.com/psf/black) to the entire file?

Reply:
Done